### ISSUE TITLE
Trainning data for wavenet_vocoder

### ISSUE THREAD

**Issue by deli666** [NVIDIA/tacotron2#46]
I trained this model with my own dataset and want to train wavenet_vocoder after this. But I am confused what is the training data for wavenet_vocoder? Mel_spectrogram generated by this model or generated by sftf?
Besides, what do you mean by "When performing Mel-Spectrogram to Audio synthesis with a WaveNet model, make sure Tacotron 2 and WaveNet were trained on the same mel-spectrogram representation"?

**Comment by harlyh** [comment]
1) While you train Tacotron, Audio is converted to Mels and the Mel+Label(text) pairs are fed to Tacotron. After successful training you should have a model, which can generate Mels from Text. 
Now for wavenet you feed the synthesised Mels+Audio samples to Wavenet. This eventually should produce a model, which is able to Synthesize Mels into ncie Audio.
So to summarise: 
* Tacotron training needs Audio+Text input and will start synthesizing Mels from input text, which you can either convert to low quality audio using Griffin-Lim, or use the Mels for next step, that is
* Wavenet training, which takes the Tacotron synthesized Mel + Audio pairs as input to produce a model which can synthesize ncie audio from Tacotron Mels.

2) It just means that if, for example, if you're trying to use pre-trained Wavenet model, then the Mel parameters used for Wavenet training must match the Mel params you used for Tacotron training (scaling, samplerate, window size, etc...)  
Or to put it differently: Mels used for Wavenet training must match the Mels used for Tacotron training, meaning that between Tacotron and Wavenet training you mustn't change Mel parameters.

Hope this explains it

**Comment by deli666** [comment]
Thank you so much for your answer!

**Comment by harlyh** [comment]
Well, haven't actually considered training Wavenet until I'm not satisfied with the Tacotron synthesis. Also I haven't tried using pre-trained Wavenet. But considering the intended workflow I do believe that Wavenet would produce the best results only when synthesising speech using the output from the same Tacotron model as was used during Wavenet training. However, I'm not saying that you can't do that. For me it just wouldn't make much sense :)

**Comment by rafaelvalle** [comment]
Please check this issue where we explain how to use mels from a Tacotron that was trained on data that does not match the same processing as wavenet.

https://github.com/NVIDIA/tacotron2/issues/41

**Comment by rafaelvalle** [comment]
Closing due to inactivity. Please re-open if necessary.
